c++/src/kj/array.h

```diff
@@ -208,11 +208,11 @@ class Array {
 
   inline Array<PropagateConst<T, byte>> releaseAsBytes() {
     // Like asBytes() but transfers ownership.
-    static_assert(sizeof(T) == sizeof(byte),
-        "releaseAsBytes() only possible on arrays with byte-size elements (e.g. chars).");
+    static_assert(KJ_HAS_TRIVIAL_CONSTRUCTOR(T),
```

Comment:
The `ArrayDisposer` expects to have the element size and count passed back to it, but if the original array had different-sized elements, then these values will end up inconsistent when passed to the disposer. That's why the check here was checking for exact size match.

Your version works correctly with `HeapArrayDisposer` because `HeapArrayDisposer` doesn't actually care about the element size, it just passes the pointer to `operator delete[]`. But in general this violates the contract of `ArrayDisposer`.

Let's just have the new function return `Array<word>` and avoid making this change.